// vnpay/dist/chunk-H67PWVQR.cjs
'use strict';

// src/enums/product-code.enum.ts
var ProductCode = /* @__PURE__ */ ((ProductCode2) => {
  ProductCode2["Food_Consumption"] = "100000";
  ProductCode2["Phone_Tablet"] = "110000";
  ProductCode2["ElectricAppliance"] = "120000";
  ProductCode2["Computers_OfficeEquipment"] = "130000";
  ProductCode2["Electronics_Sound"] = "140000";
  ProductCode2["Books_Newspapers_Magazines"] = "150000";
  ProductCode2["Sports_Picnics"] = "160000";
  ProductCode2["Hotel_Tourism"] = "170000";
  ProductCode2["Cuisine"] = "180000";
  ProductCode2["Entertainment_Training"] = "190000";
  ProductCode2["Fashion"] = "200000";
  ProductCode2["Health_Beauty"] = "210000";
  ProductCode2["Mother_Baby"] = "220000";
  ProductCode2["KitchenUtensils"] = "230000";
  ProductCode2["Vehicle"] = "240000";
  ProductCode2["Pay"] = "250000";
  ProductCode2["AirlineTickets"] = "250007";
  ProductCode2["CardCode"] = "260000";
  ProductCode2["Pharmacy_MedicalServices"] = "270000";
  ProductCode2["Other"] = "other";
  return ProductCode2;
})(ProductCode || {});

// src/enums/index.ts
var UrlService = /* @__PURE__ */ ((UrlService2) => {
  UrlService2["sandbox"] = "https://sandbox.vnpayment.vn/paymentv2/vpcpay.html";
  return UrlService2;
})(UrlService || {});
var HashAlgorithm = /* @__PURE__ */ ((HashAlgorithm2) => {
  HashAlgorithm2["SHA256"] = "SHA256";
  HashAlgorithm2["SHA512"] = "SHA512";
  HashAlgorithm2["MD5"] = "MD5";
  return HashAlgorithm2;
})(HashAlgorithm || {});
var VnpCurrCode = /* @__PURE__ */ ((VnpCurrCode2) => {
  VnpCurrCode2["VND"] = "VND";
  return VnpCurrCode2;
})(VnpCurrCode || {});
var VnpLocale = /* @__PURE__ */ ((VnpLocale2) => {
  VnpLocale2["VN"] = "vn";
  VnpLocale2["EN"] = "en";
  return VnpLocale2;
})(VnpLocale || {});
var VnpCardType = /* @__PURE__ */ ((VnpCardType2) => {
  VnpCardType2["ATM"] = "ATM";
  VnpCardType2["QRCODE"] = "QRCODE";
  return VnpCardType2;
})(VnpCardType || {});
var VnpTransactionType = /* @__PURE__ */ ((VnpTransactionType2) => {
  VnpTransactionType2["PAYMENT"] = "01";
  VnpTransactionType2["FULL_REFUND"] = "02";
  VnpTransactionType2["PARTIAL_REFUND"] = "03";
  return VnpTransactionType2;
})(VnpTransactionType || {});
var RefundTransactionType = /* @__PURE__ */ ((RefundTransactionType2) => {
  RefundTransactionType2["FULL_REFUND"] = "02";
  RefundTransactionType2["PARTIAL_REFUND"] = "03";
  return RefundTransactionType2;
})(RefundTransactionType || {});

exports.HashAlgorithm = HashAlgorithm;
exports.ProductCode = ProductCode;
exports.RefundTransactionType = RefundTransactionType;
exports.UrlService = UrlService;
exports.VnpCardType = VnpCardType;
exports.VnpCurrCode = VnpCurrCode;
exports.VnpLocale = VnpLocale;
exports.VnpTransactionType = VnpTransactionType;
