// vnpay/dist/chunk-EYO267YS.cjs
'use strict';

// src/constants/api-endpoint.constant.ts
var VNPAY_GATEWAY_SANDBOX_HOST = "https://sandbox.vnpayment.vn";
var PAYMENT_ENDPOINT = "paymentv2/vpcpay.html";
var QUERY_DR_REFUND_ENDPOINT = "merchant_webapi/api/transaction";
var GET_BANK_LIST_ENDPOINT = "qrpayauth/api/merchant/get_bank_list";

// src/constants/ipn-result-for-vnpay.constant.ts
var IpnSuccess = {
  RspCode: "00",
  Message: "Confirm Success"
};
var IpnOrderNotFound = {
  RspCode: "01",
  Message: "Order not found"
};
var InpOrderAlreadyConfirmed = {
  RspCode: "02",
  Message: "Order already confirmed"
};
var IpnIpProhibited = {
  RspCode: "03",
  Message: "IP prohibited"
};
var IpnInvalidAmount = {
  RspCode: "04",
  Message: "Invalid amount"
};
var IpnFailChecksum = {
  RspCode: "97",
  Message: "Fail checksum"
};
var IpnUnknownError = {
  RspCode: "99",
  Message: "Unknown error"
};

// src/constants/regex.constant.ts
var numberRegex = /^[0-9]+$/;

// src/constants/index.ts
var VNP_VERSION = "2.1.0";
var VNP_DEFAULT_COMMAND = "pay";
var CURR_CODE_VND = "VND";

exports.CURR_CODE_VND = CURR_CODE_VND;
exports.GET_BANK_LIST_ENDPOINT = GET_BANK_LIST_ENDPOINT;
exports.InpOrderAlreadyConfirmed = InpOrderAlreadyConfirmed;
exports.IpnFailChecksum = IpnFailChecksum;
exports.IpnInvalidAmount = IpnInvalidAmount;
exports.IpnIpProhibited = IpnIpProhibited;
exports.IpnOrderNotFound = IpnOrderNotFound;
exports.IpnSuccess = IpnSuccess;
exports.IpnUnknownError = IpnUnknownError;
exports.PAYMENT_ENDPOINT = PAYMENT_ENDPOINT;
exports.QUERY_DR_REFUND_ENDPOINT = QUERY_DR_REFUND_ENDPOINT;
exports.VNPAY_GATEWAY_SANDBOX_HOST = VNPAY_GATEWAY_SANDBOX_HOST;
exports.VNP_DEFAULT_COMMAND = VNP_DEFAULT_COMMAND;
exports.VNP_VERSION = VNP_VERSION;
exports.numberRegex = numberRegex;
