// vnpay/dist/constants.cjs
'use strict';

var chunkEYO267YS_cjs = require('./chunk-EYO267YS.cjs');
var chunkJGUV45TP_cjs = require('./chunk-JGUV45TP.cjs');
require('./chunk-UOEGPC4B.cjs');



Object.defineProperty(exports, "CURR_CODE_VND", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.CURR_CODE_VND; }
});
Object.defineProperty(exports, "GET_BANK_LIST_ENDPOINT", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.GET_BANK_LIST_ENDPOINT; }
});
Object.defineProperty(exports, "InpOrderAlreadyConfirmed", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.InpOrderAlreadyConfirmed; }
});
Object.defineProperty(exports, "IpnFailChecksum", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.IpnFailChecksum; }
});
Object.defineProperty(exports, "IpnInvalidAmount", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.IpnInvalidAmount; }
});
Object.defineProperty(exports, "IpnIpProhibited", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.IpnIpProhibited; }
});
Object.defineProperty(exports, "IpnOrderNotFound", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.IpnOrderNotFound; }
});
Object.defineProperty(exports, "IpnSuccess", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.IpnSuccess; }
});
Object.defineProperty(exports, "IpnUnknownError", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.IpnUnknownError; }
});
Object.defineProperty(exports, "PAYMENT_ENDPOINT", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.PAYMENT_ENDPOINT; }
});
Object.defineProperty(exports, "QUERY_DR_REFUND_ENDPOINT", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.QUERY_DR_REFUND_ENDPOINT; }
});
Object.defineProperty(exports, "VNPAY_GATEWAY_SANDBOX_HOST", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.VNPAY_GATEWAY_SANDBOX_HOST; }
});
Object.defineProperty(exports, "VNP_DEFAULT_COMMAND", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.VNP_DEFAULT_COMMAND; }
});
Object.defineProperty(exports, "VNP_VERSION", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.VNP_VERSION; }
});
Object.defineProperty(exports, "numberRegex", {
  enumerable: true,
  get: function () { return chunkEYO267YS_cjs.numberRegex; }
});
Object.defineProperty(exports, "QUERY_DR_RESPONSE_MAP", {
  enumerable: true,
  get: function () { return chunkJGUV45TP_cjs.QUERY_DR_RESPONSE_MAP; }
});
Object.defineProperty(exports, "REFUND_RESPONSE_MAP", {
  enumerable: true,
  get: function () { return chunkJGUV45TP_cjs.REFUND_RESPONSE_MAP; }
});
Object.defineProperty(exports, "RESPONSE_MAP", {
  enumerable: true,
  get: function () { return chunkJGUV45TP_cjs.RESPONSE_MAP; }
});
Object.defineProperty(exports, "TRANSACTION_STATUS_RESPONSE_MAP", {
  enumerable: true,
  get: function () { return chunkJGUV45TP_cjs.TRANSACTION_STATUS_RESPONSE_MAP; }
});
Object.defineProperty(exports, "WRONG_CHECKSUM_KEY", {
  enumerable: true,
  get: function () { return chunkJGUV45TP_cjs.WRONG_CHECKSUM_KEY; }
});
